synapse/handlers/user_directory.py

```diff
@@ -329,13 +333,6 @@ def _handle_deltas(self, deltas):
                     public_value=Membership.JOIN,
                 )
 
-                if change is None:
```

Comment:
I'm a bit surprised if this is a safe thing to do - can you justify why it is ok to do the stuff at lines 340-355 if `change` is None?

Reply:
The problem is that I reordered the switching logic on ```change``` so that I could combine ```None``` and ```True``` under the same ```is_support``` guard.

This means that the meaning of a line like ```if not change:``` is now different since ```change``` can legitimately be ```None``` where before this was not possible.

I've changed the if statement accordingly - thanks for spot.